lib/challenge.js

```diff
@@ -229,13 +287,53 @@ let nums = [
 ];
 
 function prompt11() {
-  let median;
+    nums.sort((a,b) => a - b); // Sort the array
+    let median; // Declare the median variable
   // YOUR CODE HERE
+  if(nums.length % 2 === 0) {//if it's even
+    median = (nums[nums.length / 2 - 1] + (nums[nums.length / 2]) / 2) ;
 
-  console.log(median);
+  } else { //it's odd 9 items / 2 would be 4.5, we would round down to 4
+    median = nums[Math.floor(nums.length/2)];
+
+  }
+  console.log(median); //Used chat GPT for assistence, need some help understanding what this is doing exactly outside of the nums.length % 2 === 0
 }
+
+prompt11();
 // COMMENT IN THE BELOW LINE OF CODE TO TEST YOUR OUTPUT IN THE BROWSER!
-// prompt11();
+
+// CHATGPT explanation of why math.floor() is used
+// const nums = [1, 2, 3, 4, 5];
+
+// nums.sort((a, b) => a - b);
+
+// let median;
+// if (nums.length % 2 === 0) {
+//   median = (nums[nums.length / 2 - 1] + nums[nums.length / 2]) / 2;
+// } else {
+//   median = nums[Math.floor(nums.length / 2)];
+// }
+
+// console.log(median); // Output: 3
+// In this example, the nums array has a length of 5, which is odd. So, in the else block, Math.floor(nums.length / 2) calculates 2.5, which is rounded down to 2. This gives us the index of the middle element in the array, which is nums[2], or 3. Since this is an odd-length array, 3 is the median, and it is logged to the console.
+
+
+//For EVEN length.
+// const nums = [1, 2, 3, 4, 5, 6];
+
+// nums.sort((a, b) => a - b);
```

Comment:
That's cool that you used chatGPT to help with the code. Do you understand this function line by line?

(This comment didn't get posted originally for some reason)